src/features/generic_events/aggregate/index.js

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright 2020 New Relic Corporation. All rights reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+import { stringify } from '../../../common/util/stringify'
+import { HarvestScheduler } from '../../../common/harvest/harvest-scheduler'
+import { cleanURL } from '../../../common/url/clean-url'
+import { getConfigurationValue, getInfo, getRuntime } from '../../../common/config/config'
+import { FEATURE_NAME } from '../constants'
+import { isBrowserScope } from '../../../common/constants/runtime'
+import { AggregateBase } from '../../utils/aggregate-base'
+import { warn } from '../../../common/util/console'
+import { now } from '../../../common/timing/now'
+
+export class Aggregate extends AggregateBase {
+  #agentRuntime
+  static featureName = FEATURE_NAME
+  constructor (agentIdentifier, aggregator) {
+    super(agentIdentifier, aggregator, FEATURE_NAME)
+
+    this.eventsPerHarvest = 1000
+    this.harvestTimeSeconds = getConfigurationValue(this.agentIdentifier, 'generic_events.harvestTimeSeconds')
+
+    this.referrerUrl = undefined
+    this.currentEvents = undefined
+
+    this.events = []
+    this.overflow = []
+
+    this.#agentRuntime = getRuntime(this.agentIdentifier)
+
+    if (isBrowserScope && document.referrer) this.referrerUrl = cleanURL(document.referrer)
+
+    this.waitForFlags(['ins']).then(([ins]) => {
+      if (ins) {
+        // handle page actions and other generic events here
+        this.harvestScheduler = new HarvestScheduler('ins', { onFinished: (...args) => this.onHarvestFinished(...args) }, this)
+        this.harvestScheduler.harvest.on('ins', (...args) => this.onHarvestStarted(...args))
+        // this.harvestScheduler.startTimer(this.harvestTimeSeconds, 0)
+      }
+      this.drain()
+    })
+  }
+
+  onHarvestStarted (options) {
+    const { userAttributes, atts } = getInfo(this.agentIdentifier)
+    const harvestEvents = this.overflow.length ? this.overflow.splice(0, Infinity) : this.events.splice(0, Infinity)
+    var payload = ({
+      qs: {
+        ua: userAttributes,
+        at: atts
+      },
+      body: {
+        ins: harvestEvents
+      }
+    })
+
+    if (options.retry) {
+      this.currentEvents = harvestEvents
+    }
+
+    return payload
+  }
+
+  onHarvestFinished (result) {
+    if (result && result.sent && result.retry && this.currentEvents) {
+      this.events = this.events.concat(this.currentEvents)
```

Comment:
currentEvents should maintain first in order

Reply:
why does this matter?  Every event is timestamped

Reply:
just the pattern with all the other feature like trace; it's also implied when we troubleshoot and open payloads in dev tools to find the earliest at the top